src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
+            idx = [indexin(a.pool.index, levels) for a in As]
+
+            ordered = A1.pool.ordered
```

Comment:
Result should be ordered only if all inputs are. Also, use high-level methods where possible: `ordered(A1)`.